nearcore/src/config.rs

```diff
@@ -1181,15 +1184,36 @@ pub fn create_testnet_configs_from_seeds(
         .iter()
         .map(|seed| InMemorySigner::from_seed("node".parse().unwrap(), KeyType::ED25519, seed))
         .collect::<Vec<_>>();
-    let genesis = Genesis::test_sharded(
-        seeds.iter().map(|s| s.parse().unwrap()).collect(),
+
+    let shard_layout = if let Some(ref fixed_shards) = fixed_shards {
+        // If fixed shards are set, we expect that they take over all the shards (except for one that would host all the other accounts).
+        assert!(fixed_shards.len() == num_shards as usize - 1);
+        ShardLayout::v1(
+            fixed_shards.iter().map(|it| it.parse().unwrap()).collect(),
+            vec![],
+            None,
+            0,
+        )
+    } else {
+        ShardLayout::v0(num_shards, 0)
+    };
+    // If we have fixed shards - let's also add those accounts to genesis.
+    let accounts_to_add_to_genesis = if let Some(ref fixed_shards_accounts) = fixed_shards {
+        seeds.iter().chain(fixed_shards_accounts.iter()).map(|s| s.parse().unwrap()).collect()
+    } else {
+        seeds.iter().map(|s| s.parse().unwrap()).collect()
+    };
```

Comment:
This might be cleaner as two steps, collect and than conditional expand.
```rust
let accounts_to_add = seeds.iter().map(...).collect();
if let Some(ref accounts) = fixed_shards {
    accounts_to_add.expand(accounts.iter().map(...));
}
```

Reply:
ok